flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/RecordContext.java

```diff
@@ -56,6 +59,9 @@ public class RecordContext<K> extends ReferenceCounted<RecordContext.DisposerRun
     /** The keyGroup to which key belongs. */
     private final int keyGroup;
 
+    /** The namespaces of states. Lazy initialization for saving memory. */
```

Comment:
Will there be any concurrency problem caused by ` Lazy initialization`?

Reply:
Nope. The `setNamespace` is only invoked on task thread.